code/game/machinery/computer/arcade.dm

```diff
@@ -431,34 +453,45 @@
 	if(href_list["close"])
 		usr.unset_machine()
 		usr << browse(null, "window=arcade")
-	else if (href_list["continue"]) //Continue your travels
-		if(turns >= 9)
+
+	if(busy)
+		return
+	busy = 1
+
+	if (href_list["continue"]) //Continue your travels
+		if(turns >= ORION_TRAIL_WINTURN)
 			win()
 		else if(turns == 2)
 			if(prob(30))
-				event = "Collision"
+				event = ORION_TRAIL_COLLISION
 				event()
-				food -= alive*2
+				food -= (alive+lings_aboard)*2
 				fuel -= 5
 				turns += 1
 			else
-				food -= alive*2
+				food -= (alive+lings_aboard)*2
 				fuel -= 5
 				turns += 1
 				if(prob(75))
 					event = pickweight(events)
+					if(lings_aboard)
+						if(event == ORION_TRAIL_LING || prob(55))
+							event = ORION_TRAIL_LING_ATTACK
 					event()
 		else
-			food -= alive*2
+			food -= (alive+lings_aboard)*2
 			fuel -= 5
 			turns += 1

```

Comment:
You can avoid having those three lines copypasted three times by changing `else if(turn ==2)` into
```
else
	food -= (alive+lings_aboard)*2
 	fuel -= 5
 	turns += 1
	if(turns == 2)
		if(prob(30))
		else
	else
		if(prob(75))